taskvine/src/bindings/python3/ndcctools/taskvine/task.py

```diff
@@ -1059,12 +1047,34 @@ def __init__(self, library, fn, *args, **kwargs):
         # function calls at worker only need the name of the function.
         self.set_command(fn)
 
+        self._library = library
+        self._library_name = self.get_library_name()
+        self._function_name = fn
+
         self._event = {}
         self._event["fn_args"] = args
         self._event["fn_kwargs"] = kwargs
 
         self._saved_output = None
-        self.set_library_required(library)
+        self.set_library_required(self._library_name)
+
+    def get_library_name(self):
```

Comment:
This should simply return `self._library_name`. In general we use `get_` to obtain the value of attributes only.

Reply:
Instead, convert to `_compute_library_name(...)`, which should set `self._library_name`.